src/views/search/utils.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-FileCopyrightText: 2021 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { TFunction } from 'i18next';
+import { v4 as uuid } from 'uuid';
+
+import { ContactInputItem, KeywordState, Query, SearchQueryItem } from '../../types';
+
+type EmptyListMessages = { title: string; description: string }[];
+
+export const EmptyListMessages = (t: TFunction): EmptyListMessages => [
+	{
+		title: t('displayer.search_title1', 'Start another search'),
+		description: t(
+			'displayer.search_description1',
+			'Or select “Advanced Filters” to refine your search.'
+		)
+	},
+	{
+		title: t('displayer.search_title2', 'We’re sorry but there are no results for your search'),
+		description: t('displayer.search_description2', 'Try to start another search.')
+	},
+	{
+		title: t('displayer.search_title3', 'There are no results for your search.'),
+		description: t(
+			'displayer.search_description3',
+			'Check the spelling and the filters options or try with another keyword.'
+		)
+	}
+];
+
+export const EmptyFieldMessages = (t: TFunction): EmptyListMessages => [
+	{
+		title: t(
+			'displayer.search_title4',
+			'Select one or more results to perform actions or display details.'
+		),
+		description: ''
+	}
+];
+
+function getRegex(prefix?: string): RegExp {
+	return new RegExp(`^${prefix}:.*`, 'i');
+}
+
+export function getChipString(item: SearchQueryItem | ContactInputItem, prefix: string): string {
+	const regex = getRegex(prefix);
+	let resultString = '';
+	if ((item as SearchQueryItem).label) {
+		resultString = (item as SearchQueryItem).label;
+	}
+	if ((item as ContactInputItem).fullName) {
+		resultString = (item.label || (item as ContactInputItem).fullName) ?? '';
+	}
+	return regex.test(resultString) ? resultString : `${prefix}:${resultString}`;
+}
+
+export function getChipItems(chips: Query | Array<ContactInputItem>, prefix: string): KeywordState {
+	return chips.map((chip) => ({
+		...chip,
+		error: false,
+		id: chip.id ?? `${uuid()} ${chip.label}`,
+		avatarBackground: (chip as ContactInputItem).avatarBackground ?? 'secondary',
+		hasAvatar: true,
+		avatarIcon: 'EmailOutline',
+		isGeneric: false,
+		isQueryFilter: true,
+		label: getChipString(chip, prefix),
+		fullName: getChipString(chip, prefix),
+		value: getChipString(chip, prefix)
```

Comment:
Compared to the previous implementation the value now will contain the full name  instead of the email address: this breaks the search which expects an email address

Reply:
I don't understand what you mean, I can see the implementation working correctly as is. 
I deployed it on https://co-dev-pry3.demo.zextras.io/ if you want to take a look.
in the to and from fields the search expects a prefix and an email address, and that's what it is receiving.
am i missing a case in which this implementation does not work?

Reply:
ok, i see what you mean. but that does work:

https://wiki.zimbra.com/wiki/Zimbra_Web_Client_Search_Tips

under keyword description and examples:

from: Specifies a sender name or email address that is in the From header. This can be text, as in "John Smith III", an email address such as "joe@acme.com", or a domain such as "@zimbra.com".

Reply:
I know that the BE is accepting also something else than the address, but this change is altering the behaviour of the search:
- if the message that the user is searching for is sent without the sender name, the user now is not able to find it
- if the sender name is set in a message sent from another domain (like the notification that we receive from Confluence) the user is getting different results, compared to the previous implementation.
In general I think we should limit the changes to just fix the bug

Reply:
I see, the issue is actually on the "value" field, which should receive either the email from the ContactInputItem type or the value from the SearchQueryItem. I'm going to change that